endpoint.php

```diff
@@ -16,35 +16,40 @@
  *
  * Requirements:
  *  - PHP 5.3 or newer
- *  - Amazon PHP SDK (only if utilizing the AWS SDK for deleting files or otherwise examining them)
+ *  - Amazon PHP SDK v3 (only if utilizing the AWS SDK for deleting files or otherwise examining them)
  *
- * If you need to install the AWS SDK, see http://docs.aws.amazon.com/aws-sdk-php-2/guide/latest/installation.html.
+ * If you need to install the AWS SDK, see https://docs.aws.amazon.com/aws-sdk-php/v3/guide/getting-started/installation.html .
+ *
+ * Updated to AWS PHP SDK as per https://docs.aws.amazon.com/aws-sdk-php/v3/guide/guide/migration.html
+ *
+ * You will need to specify your S3 region in objectProperties, and uploadSuccess params specify region and version
  */
+ require '../../autoload.php';
+ use Aws\S3\S3Client;
+
+ // These assume you have the associated AWS keys stored in
+ // the associated system environment variables
+ $clientPrivateKey = $_ENV['AWS_CLIENT_SECRET_KEY'];
+ // These two keys are only needed if the delete file feature is enabled
+ // or if you are, for example, confirming the file size in a successEndpoint
+ // handler via S3's SDK, as we are doing in this example.
+ $serverPublicKey = $_ENV['AWS_SERVER_PUBLIC_KEY'];
+ $serverPrivateKey = $_ENV['AWS_SERVER_PRIVATE_KEY'];
+
+
+ // The following variables are used when validating the policy document
+ // sent by the uploader.
+ $expectedBucketName = $_ENV['S3_BUCKET_NAME'];
+ $expectedHostName = $_ENV['S3_HOST_NAME']; // v4-only
```

Comment:
Here should be added `if` statement for v4 only requests. Because it will trigger Notice about empty array value